components/brave_wallet/browser/eth_response_parser.cc

```diff
@@ -249,14 +254,36 @@ bool ParseUnstoppableDomainsProxyReaderGetMany(
 
 bool ParseUnstoppableDomainsProxyReaderGet(const std::string& json,
                                            std::string* value) {
-  DCHECK(value);
+  return ParseStringResult(json, value);
+}
 
+bool ParseERC721TokenUri(const std::string& json, GURL* url) {
   std::string result;
-  if (!ParseSingleStringResult(json, &result))
+  if (!ParseStringResult(json, &result)) {
     return false;
+  }
 
-  size_t offset = 2 /* len of "0x" */ + 64 /* len of offset to array */;
-  return brave_wallet::DecodeString(offset, result, value);
+  GURL result_url = GURL(result);
+  if (!result_url.is_valid()) {
```

Comment:
should this also return false if result_url is an unexpected scheme?

Reply:
Currently doing that outside of this function here https://github.com/brave/brave-core/blob/d917df512f60cb0e1c1654f3c95ee3838d5aea66/components/brave_wallet/browser/json_rpc_service.cc#L1613-L1619
The benefit of doing it that way is that we can pass an 'not supported' type error, vs. the default in ParseErrorResult which is a 'parsing error' if ParseERC721TokenUri returns false.